File: MultiDimensionalArray/ScoreAfterFlippingMatrix/scoreAfterFlipMatrix.py
from typing import List

class Solution:
    def matrixScore(self, grid: List[List[int]]) -> int:
        rows = len(grid)
        cols = len(grid[0])
        for i in range(rows):
            if not grid[i][0]:
                for j in range(cols):
                    if grid[i][j]: grid[i][j] = 0
                    else: grid[i][j] = 1
        for j in range(cols):
            noz = 0
            for i in range(rows):
                if not grid[i][j]: noz += 1
            if noz > (rows - noz):
                for i in range(rows):
                    if grid[i][j]: grid[i][j] = 0
                    else: grid[i][j] = 1
        ans = 0
        for i in range(rows):
            value = 0
            mul = 1
            for j in range(cols-1,-1,-1):
                value += (grid[i][j]*mul)
                mul *= 2
            ans += value
        return ans            
        

File: MultiDimensionalArray/ScoreAfterFlippingMatrix/scoreAfterFlipMatrix.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int matrixScore(vector<vector<int>>& grid) {
        int rows = grid.size();
        int cols = grid[0].size();
        // Flip Rows
        for(int i=0;i<rows;i++){
            if(!grid[i][0]){
                for(int j=0;j<cols;j++){
                    if(grid[i][j])grid[i][j] = 0;
                    else grid[i][j] = 1;
                }
            }
        }
        // Flip Cols
        for(int j=0;j<cols;j++){
            int noz = 0;
            for(int i=0;i<rows;i++){
                if(!grid[i][j])noz++;
            }
            if(noz>rows-noz){
                for(int i=0;i<rows;i++){
                    if(grid[i][j])grid[i][j] = 0;
                    else grid[i][j] = 1;
                }    
            }
        }
        int ans = 0;
        for(int i=0;i<rows;i++){
            int value = 0;
            int mul = 1;
            for(int j=cols-1;j>=0;j--){
                value += (grid[i][j]*mul);
                mul *= 2;
            }
            ans += value;
        }
        return ans;
    }
};